python/graphstorm/config/argument.py

```diff
@@ -1215,21 +1218,98 @@ def input_activate(self):
+
+        This method parses given edge feature name list, and return either a string
+        corresponding a global feature name, or a dictionary corresponding different
+        edge types with diffent feature names.
         """
+        # pylint: disable=no-member
+        if hasattr(self, "_edge_feat_name"):
+            feat_names = self._edge_feat_name
+            if len(feat_names) == 1 and \
+                ":" not in feat_names[0]:
+                # global feat_name
+                return feat_names[0]
+
+            # per edge type feature
+            fname_dict = {}
+
+            for feat_name in feat_names:
+                feat_info = feat_name.split(":")
+                assert len(feat_info) == 2, \
+                        f"Unknown format of the feature name: {feat_name}, " + \
+                        "must be: can_etype:feat_name."
+                # check and convert canonical edge type string
+                assert isinstance(feat_info[0], str), \
+                    f"The edge type should be a string not {feat_info[0]}"
+                can_etype = tuple(item.strip() for item in feat_info[0].split(","))
+                assert len(can_etype) == 3, \
+                        f"Unknown format of the edge type {feat_info[0]}, must be: " + \
+                         "src_node_type,relation_type,dst_node_type."
+                assert can_etype not in fname_dict, \
+                        f"You already specify the feature names of {can_etype} " \
+                        f"as {fname_dict[can_etype]}."
+                assert isinstance(feat_info[1], str), \
+                    f"Feature name of {can_etype} should be a string not {feat_info[1]}"
+                # multiple features separated by ','
+                fname_dict[can_etype] = [item.strip() for item in feat_info[1].split(",")]
+            return fname_dict
+
+        # By default, return None which means there is no node feature
         return None
 
+    @property
+    def edge_feat_mp_op(self):
+        """ The operation for using edge features during message passing computation.
+            Defaut is "concat".
+
+        .. versionadded:: 0.4.0
+            The ``edge_feat_mp_op`` argument.
+
+            GraphStorm supports five message passing operations for edge features, including:
+
+            - "concat": concatinate the source node feature with the edge feauture together,
+              and then pass them to the destination node.
+            - "add":add the source node feature with the edge feauture together,
+              and then pass them to the destination node.
+            - "sub":substract the edge feauture from the source node feature,
+              and then pass them to the destination node.
+            - "mul":multiple the source node feature with the edge feauture,
+              and then pass them to the destination node.
+            - "div":divid the source node feature by the edge feauture together,
```

Comment:
Do we need div and sub? They are similar to add and div.

Reply:
I asked a few users if they can implement the `sub` and `div` on their own. They have no idea how to do this. Even I told them to modify the input feature, they think this is not flexible as we need to run gconstruct every time after modify edge feature.

Because implement this is simple, and users only need to configure with a string, I decide to add the two operations.